examples/with-mysql/docker-compose.yml

```diff
@@ -1,42 +1,84 @@
+  # Adjust for what is feasible for your setup
+  deploy:
+    resources:
+      limits:
+        cpus: '1'
+        memory: 512M
+      reservations:
+        cpus: '0.2'
+        memory: 32M
+  volumes:
+    - custom:/var/www/html/custom
+    - documents:/var/www/documents
+  restart: unless-stopped
+
+volumes:
+  custom:
+  documents:
+  db:
+
 networks:
   internal-pod:
-    internal: true
   external-pod:
-    internal: false
-
-volumes:
-  dolibarr-custom:
-  dolibarr-docs:
-  mysql-data:
 
 services:
-  dolibarr:
-    image: dolibarr/dolibarr:latest
+  db:
+    image: mysql:lts
+    # Limiting the container's resources isn't mandatory, but it's recommended.
+    # Adjust for what is feasible for your setup
+    deploy:
+      resources:
+        limits:
+          cpus: '1'
+          memory: 512M
+        reservations:
+          cpus: '0.2'
+          memory: 32M
+    healthcheck:
+      test: ["CMD", "healthcheck.sh", "--connect", "--innodb_initialized"]
+      start_period: 60s
+      interval: 10s
+      timeout: 5s
+      retries: 3
     environment:
-      DOLI_DB_HOST: "mysql"
-      DOLI_DB_HOST_PORT: "3306"
-      DOLI_DB_USER: "dolidbuser"
-      DOLI_DB_PASSWORD: "dolidbpass"
-      DOLI_DB_NAME: "dolibarr"
-      DOLI_ADMIN_LOGIN: "admin"
-      DOLI_ADMIN_PASSWORD: "mysuperhypersecretpasswordforadminacount"
-    ports:
-      - 80:80
+      MYSQL_DATABASE: ${DOLI_DB_NAME:-dolidb}
+      MYSQL_USER: ${DOLI_DB_USER:-dolidbuser}
+      MYSQL_PASSWORD: ${DOLI_DB_PASSWORD:-dolidbpass}
+      MYSQL_ROOT_PASSWORD: $ROOT_DB_PASSWORD
     volumes:
-      - dolibarr-docs:/var/www/documents
-      - dolibarr-custom:/var/www/html/custom
+      - db:/var/lib/mysql
     networks:
       - internal-pod
-      - external-pod
 
-  mysql:
-    image: mysql:latest
+  app:
+    <<: *dolibarr
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://127.0.0.1"]
+      start_period: 300s
+      interval: 30s
+      timeout: 10s
+      retries: 3
+    depends_on:
+      db:
+        condition: service_healthy
     environment:
-      MYSQL_DATABASE: "dolibarr"
-      MYSQL_USER: "dolibarr"
-      MYSQL_PASSWORD: "mysupersecretpasswordfordatabase"
-      MYSQL_ROOT_PASSWORD: "mysupersupersecretpasswordforrootuser"
-    volumes:
-      - mysql-data:/var/lib/mysql
+      WWW_USER_ID: ${WWW_USER_ID:-1000}
+      WWW_GROUP_ID: ${WWW_GROUP_ID:-1000}
+      DOLI_DB_HOST: db
+      DOLI_DB_NAME: ${DOLI_DB_NAME:-dolidb}
+      DOLI_DB_USER: ${DOLI_DB_USER:-dolidbuser}
+      DOLI_DB_PASSWORD: ${DOLI_DB_PASSWORD:-dolidbpass}
+      DOLI_URL_ROOT: "${DOLI_URL_ROOT:-http://0.0.0.0}"
+      DOLI_ADMIN_LOGIN: "${DOLI_ADMIN_LOGIN:-admin}"
+      DOLI_ADMIN_PASSWORD: "${DOLI_ADMIN_PASSWORD:-admin}"
+      DOLI_INIT_DEMO: ${DOLI_INIT_DEMO:-0}
+      DOLI_COMPANY_NAME: ${DOLI_COMPANY_NAME:-MyBigCompany}
+      DOLI_INSTANCE_UNIQUE_ID: $DOLI_INSTANCE_ID # Add a random ASCII 16 char string here
```

Comment:
I would remove the comment so that the user perform the change in the .env directly